### PR TITLE
Add missing java api for StreamTestKit

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
Add more java native api to StreamTestKit.

refs:https://github.com/apache/incubator-pekko/issues/970

was: https://github.com/apache/incubator-pekko/pull/1049

continue the work of @naosense

### CODE REVIEW

File: stream-testkit/src/test/scala/org/apache/pekko/stream/testkit/StreamTestKitSpec.scala [hunk 1]
```diff
@@ -136,15 +136,15 @@ class StreamTestKitSpec extends PekkoSpec {
         // It also needs to be dilated since the testkit will dilate the timeout
         // accordingly to `-Dpekko.test.timefactor` value.
         val initialDelay = (timeout * 2).dilated
+        val pf: PartialFunction[Any, Unit] = {
```
Comment (pjfanning) [inline stream-testkit/src/test/scala/org/apache/pekko/stream/testkit/StreamTestKitSpec.scala:139]: these changes seem quite small compared to the StreamTestKit changes

is it possible to test all the new functions?

Comment (He-Pin) [reply stream-testkit/src/test/scala/org/apache/pekko/stream/testkit/StreamTestKitSpec.scala:139]: Sorry, I'm a little busy at $Work right now, testing all these new methods seems like a lot of work, and as we can see, the changes are mostly small, just forwarding/converting.

File: stream-tests/src/test/java/org/apache/pekko/stream/javadsl/SourceTest.java [hunk 2]
```diff
@@ -1242,8 +1242,7 @@ public void mustBeAbleToUseMerge3() {
             .mergeAll(Arrays.asList(sourceB, sourceC), false)
             .runWith(TestSink.probe(system), system);
     sub.expectSubscription().request(9);
-    sub.expectNextUnorderedN(Util.immutableSeq(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9)))
-        .expectComplete();
+    sub.expectNextUnorderedN(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9)).expectComplete();
```
Comment (He-Pin) [inline stream-tests/src/test/java/org/apache/pekko/stream/javadsl/SourceTest.java:1245]: Then, no `Util.immutableSeq` is needed

**Review by mdedetrich — APPROVED** [review]
This is getting a lgtm. Adding tests for these functions are not necessary since they are basic wrappers over existing functions which are already being tested and this has also historically been the case for Akka as well

**Comment by He-Pin** [comment]
@pjfanning @Roiocam WDYT, should it be included in m1 release？

**Comment by pjfanning** [comment]
Lgtm

**Review by pjfanning — APPROVED** [review]
Lgtm

**Comment by He-Pin** [comment]
Thanks

**Pull request merged**